src/inmanta/deploy/tasks.py

```diff
@@ -45,13 +46,21 @@ class Task(abc.ABC):
     """
 
     resource: ResourceIdStr
-
     id: resources.Id = dataclasses.field(init=False, compare=False, hash=False)
+    # This field is optional because task are sometimes solely created to see if an entry exists in different structure
+    # In these cases, we don't know the reason of the temporary "task"
+    reason: typing.Optional[str] = dataclasses.field(compare=False, hash=False, default=None)
 
     def __post_init__(self) -> None:
         # use object.__setattr__ because this is a frozen dataclass, see dataclasses docs
         object.__setattr__(self, "id", resources.Id.parse_id(self.resource))
 
+    def set_reason(self, reason: str) -> None:
+        # use object.__setattr__ because this is a frozen dataclass, see dataclasses docs
+        # This does not impact the __hash__ method of the dataclass because this field is not taken into consideration when
+        # computing the hash of the object (see dataclass field attributes)
+        object.__setattr__(self, "reason", reason)
```

Comment:
if it is frozen, it is frozen, no hacking around it.

Reply:
I agree.